uisdk/src/main/java/tech/dojo/pay/uisdk/presentation/ui/carddetailscheckout/state/CardDetailsCheckoutState.kt

```diff
@@ -1,36 +1,76 @@
 package tech.dojo.pay.uisdk.presentation.ui.carddetailscheckout.state
 
-import androidx.annotation.StringRes
 import tech.dojo.pay.uisdk.presentation.ui.carddetailscheckout.entity.SupportedCountriesViewEntity
 
 internal data class CardDetailsCheckoutState(
-    val totalAmount: String,
-    val amountCurrency: String,
-    val allowedPaymentMethodsIcons: List<Int>,
-    val cardHolderInputField: InputFieldState,
-    val emailInputField: InputFieldState,
-    val isEmailInputFieldRequired: Boolean,
-    val cardNumberInputField: InputFieldState,
-    val cardExpireDateInputField: InputFieldState,
-    val cvvInputFieldState: InputFieldState,
-    val isBillingCountryFieldRequired: Boolean,
-    val supportedCountriesList: List<SupportedCountriesViewEntity>,
-    val currentSelectedCountry: SupportedCountriesViewEntity,
-    val isPostalCodeFieldRequired: Boolean,
-    val postalCodeField: InputFieldState,
-    val saveCardCheckBox: CheckBoxItem,
-    val isLoading: Boolean,
-    val isEnabled: Boolean
+    var isLoading: Boolean = false,
+    var toolbarTitle: String = "",
+    var headerType: CardCheckOutHeaderType = CardCheckOutHeaderType.AMOUNT_HEADER,
+    var orderId: String? = null,
+    var merchantName: String? = null,
+    var totalAmount: String = "",
+    var amountCurrency: String = "",
+    var allowedPaymentMethodsIcons: List<Int> = listOf(),
+    var cardHolderInputField: InputFieldState = InputFieldState(value = ""),
+    var emailInputField: InputFieldState = InputFieldState(value = ""),
+    var isEmailInputFieldRequired: Boolean = false,
+    var cardNumberInputField: InputFieldState = InputFieldState(value = ""),
+    var cardExpireDateInputField: InputFieldState = InputFieldState(value = ""),
+    var cvvInputFieldState: InputFieldState = InputFieldState(value = ""),
+    var isBillingCountryFieldRequired: Boolean = false,
+    var supportedCountriesList: List<SupportedCountriesViewEntity> = listOf(),
+    var currentSelectedCountry: SupportedCountriesViewEntity = SupportedCountriesViewEntity(
+        "",
+        "",
+        false,
+    ),
+    var isPostalCodeFieldRequired: Boolean = false,
+    var postalCodeField: InputFieldState = InputFieldState(value = ""),
+    var checkBoxItem: CheckBoxItem = CheckBoxItem(
+        isVisible = false,
+        isChecked = false,
+        messageText = "",
+    ),
+    var actionButtonState: ActionButtonState = ActionButtonState(),
```

Comment:
can it be break by less parameters with grouping responsability between them? for example: 
```kotlin
CardState(
    var cardHolderInputField: InputFieldState = InputFieldState(value = ""),
    var cardNumberInputField: InputFieldState = InputFieldState(value = ""),
    var cardExpireDateInputField: InputFieldState = InputFieldState(value = ""),
    var cvvInputFieldState: InputFieldState = InputFieldState(value = ""),
)
Inputs(
    var emailInputField: InputFieldState = InputFieldState(value = ""),
    var isEmailInputFieldRequired: Boolean = false,
    ...
)
```
(Don't use the names I gave, it was an example)
This would improve a lot the readability.

Reply:
this will do lots of changes in the ViewModel I can add this as part on the optimizations on the back log